src/lib/UserSetting/UserSettingService.php

```diff
@@ -99,21 +133,19 @@ public function countUserSettings(): int
         return $this->valueRegistry->countValueDefinitions();
     }
 
-    /**
-     * @param \Ibexa\Contracts\User\UserSetting\ValueDefinitionInterface[] $values
-     * @param array $userPreferences
-     *
-     * @return \Ibexa\User\UserSetting\UserSetting[]
-     */
-    private function createUserSettings(array $values, array $userPreferences): array
+    private function createUserSettings(string $groupId, ValueDefinitionGroupInterface $group, array $userPreferences): UserSettingGroup
```

Comment:
You can improve readability:
```suggestion
    private function createUserSettings(
        string $groupId, 
        ValueDefinitionGroupInterface $group, 
        array $userPreferences
   ): UserSettingGroup {
```

Also what I don't really like here is the fact that the method is called `createUserSettings` but it returns `UserSettingsGroup` object. To me it feels more natural to call it `createUserSettingsGroup`.